cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -87,163 +134,282 @@ namespace detail {
             bool wasResponseSent = _responseSentFlag.test_and_set();
             if (!wasResponseSent)
             {
-                _self = this->shared_from_this();
                 _responder.FinishWithError(status, static_cast<void*>(this));
             }
         }
 
         void invoke(bool /* ok */) override
         {
-            // response has been sent, so we no longer need to keep
-            // ourselves alive
-            _self.reset();
+            // The response has been sent, so we no longer need to keep
+            // ourselves alive: release the implicit initial refcount that
+            // this instance was constructed with.
+            Release(ReleaseTrySend::No);
         }
     };
 
-}
+    template <typename TRequest, typename TResponse>
+    void intrusive_ptr_add_ref(unary_call_impl<TRequest, TResponse>* call)
+    {
+        call->AddRef();
+    }
 
-/// @brief The details of a single async, unary call.
-///
-/// Call \p Finish or \p FinishWithError to send a response back to the
-/// client.
-///
-/// If no explicit call to Finish/FinishWithError has been made before this
-/// unary_call instance is destroyed, a generic internal server error is
-/// sent.
-///
-/// @note This class can only be moved.
-template <typename TRequest, typename TResponse>
-class unary_call final
-{
-public:
-    unary_call(unary_call&& other) noexcept
-        : _impl(std::move(other._impl))
+    template <typename TRequest, typename TResponse>
+    void intrusive_ptr_release(unary_call_impl<TRequest, TResponse>* call)
     {
-        other._impl.reset();
+        call->Release(unary_call_impl<TRequest, TResponse>::ReleaseTrySend::Yes);
     }
 
-    unary_call& operator=(unary_call&& rhs)
+    /// @brief Detail class that helps implement \ref unary_call and \ref
+    /// shared_unary_call.
+    template <typename TRequest, typename TResponse>
+    class unary_call_base
     {
-        if (this != &rhs)
+    private:
+        using impl_type = unary_call_impl<TRequest, TResponse>;
+
+    public:
+        unary_call_base() = default;
```

Comment:
I don't think we need to explicitly `=default` everything, especially there are no custom constructors. You might want to put the default constructed under `protected` section.